components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/EndpointCertificatesApiServiceImpl.java

```diff
@@ -226,6 +233,53 @@ public Response updateEndpointCertificateByAlias(String alias, InputStream certi
         return null;
     }
 
+    @Override
+    public Response getCertificateUsageByAlias(String alias, Integer limit, Integer offset, MessageContext messageContext) throws APIManagementException {
+
+        List<API> allMatchedApis = new ArrayList<>();
+        APIQuickInfoListDTO apiQuickInfoListDTO;
+        limit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
+        offset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
+        CertificateMetadataDTO certificateMetadataDTO;
+
+        try {
+
+            if (StringUtils.isEmpty(alias)) {
+                RestApiUtil.handleBadRequest("The alias should not be empty", log);
+            }
+
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            certificateMetadataDTO = apiProvider.getCertificate(alias);
+            String endpoint = certificateMetadataDTO.getEndpoint();
+
+            if(endpoint.contains(URL_PROTOCOL_SEPARATOR)){
+                String[] parts = endpoint.split(URL_PROTOCOL_SEPARATOR);
+                endpoint = parts[parts.length-1];
+            }
+
+            String query = ENDPOINT_CONFIG_SEARCH_TYPE_PREFIX + ":" + endpoint;
+            String organization = RestApiUtil.getValidatedOrganization(messageContext);
+            Map<String, Object> searchResult = apiProvider.searchPaginatedAPIsByFQDN(query, organization, offset, limit);
+            Set<API> apis = (Set<API>) searchResult.get("apis");
+            allMatchedApis.addAll(apis);
+            apiQuickInfoListDTO = (APIQuickInfoListDTO) APIMappingUtil.fromAPIListToAPIQuickInfoListDTO(allMatchedApis);
+
+            //Add pagination section in the response
+            Object totalLength = searchResult.get("length");
+            Integer length = 0;
+            if (totalLength != null) {
+                length = (Integer) totalLength;
+            }
+
+            APIMappingUtil.setPaginationParamsForAPIQuickInfoListDTO(apiQuickInfoListDTO, query, offset, limit, length);
+            return Response.status(Response.Status.OK).entity(apiQuickInfoListDTO).build();
+
+        } catch (APIManagementException e) {
```

Comment:
what will be the response if an invalid alias provided?

Reply:
Empty list